pkg/secrethub/auth_aws.go

```diff
@@ -0,0 +1,62 @@
+package secrethub
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/auth"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+)
+
+type awsAuthService struct {
+	client client
+}
+
+func newAWSAuthService(client client) AuthMethodService {
+	return &awsAuthService{
+		client: client,
+	}
+}
+
+func (s awsAuthService) Authenticate() (auth.Authenticator, error) {
+	// Currently always use the eu-west-1 region.
+	region := endpoints.EuWest1RegionID
+
+	cfg := aws.NewConfig().WithRegion(region).WithEndpoint("sts." + region + ".amazonaws.com")
+	awsSess, err := session.NewSession(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("could not get AWS session: %v", err)
+	}
+
+	svc := sts.New(awsSess, cfg)
+
+	stsReq, _ := svc.GetCallerIdentityRequest(&sts.GetCallerIdentityInput{})
+
+	err = stsReq.Sign()
+	if err != nil {
+		return nil, fmt.Errorf("could not sign STS request: %v", err)
+	}
+
+	var buf bytes.Buffer
+	err = stsReq.HTTPRequest.Write(&buf)
+	if err != nil {
+		return nil, err
+	}
+
+	req := api.NewAuthRequestAWSSTS(api.SessionTypeHMAC, region, buf.Bytes())
+	resp, err := s.client.httpClient.Authenticate(req)
```

Comment:
Should this be `GetSession` or `CreateSession`?

Reply:
From a client-perspective that makes sense. But what would that look like on the API? It should be a POST request, but a `POST /v1/session` doesn't look right. So should we stay with `POST /v1/auth`?

Reply:
`POST /v1/sessions` makes quite some sense to me. Then you could also (later) build `GET /v1/sessions` to see all your currently logged in devices or something similar. 

Then `CreateSession` makes a lot of sense for the function name. If we ever want to introduce the concept of sessions as an API resource (e.g. similar to a repository being a resource) instead of an authentication thingy, we could now decide to put the `/sessions` route under e.g. `/v1/auth/sessions` or `/v1/me/sessions` so it won't conflict later.

I don't foresee this being a problem anytime soon though.

Reply:
`POST /v1/sessions` is not correct from a REST point of view. It means: "I have a session here, save it for me.". But that's not what we're doing. You're sending authentication data and retrieving a newly created session.

What about the following?
`POST /v1/auth` -> authenticate and create a session
`GET /v1/auth/sessions` -> returns all user's sessions.
`DELETE /v1/auth/sessions/<id>` -> retract a session

Reply:
@mackenbach, can I resolve this?

Reply:
Yes `/v1/auth/sessions` as an endpoint seems fine to me. Whether you then use `/v1/auth` or `/v1/auth/sessions` for the POST request is a bit preference. One thing to think about is that we may want to add `/v1/auth/foo_method` or something later, so consider how that impacts adding the POST on the `/v1/auth` endpoint.

Reply:
My comment was mostly about the client name, which I think was renamed to `CreateSession` or something like that. So if that's correct, you can resolve this discussion as far as I'm concerned.